shell/browser/api/electron_api_web_frame_main.cc

```diff
@@ -56,36 +56,54 @@ namespace electron {
 
 namespace api {
 
-typedef std::unordered_map<content::RenderFrameHost*, WebFrameMain*>
-    RenderFrameMap;
-base::LazyInstance<RenderFrameMap>::DestructorAtExit g_render_frame_map =
-    LAZY_INSTANCE_INITIALIZER;
+typedef std::unordered_map<int, WebFrameMain*> WebFrameMainIdMap;
 
-WebFrameMain* FromRenderFrameHost(content::RenderFrameHost* rfh) {
-  auto frame_map = g_render_frame_map.Get();
-  auto iter = frame_map.find(rfh);
+WebFrameMainIdMap& GetWebFrameMainMap() {
+  static base::NoDestructor<WebFrameMainIdMap> instance;
+  return *instance;
+}
+
+// static
+WebFrameMain* WebFrameMain::FromFrameTreeNodeId(int frame_tree_node_id) {
+  auto frame_map = GetWebFrameMainMap();
```

Comment:
```suggestion
  WebFrameMainIdMap& frame_map = GetWebFrameMainMap();
```

please! Prefer to avoid `auto` unless it's either (a) blindingly obvious what the type of the variable is, or (b) a really verbose type, like iterators.